### PR TITLE
MOBILE-19: iOS SDK improvement

### PR DESCRIPTION
**Pull request by Merthur**
MOBILE-19: 
1. Add detection and get_all_fl parameters;
2. Revise the original example to make it workable on iOS 9.1 (See NewExample folder).

Reviewers: @yulu @jasonpeng

### CODE REVIEW

File: ViSearchSDK/ViSearchAPISource/BaseSearchParams.h [hunk 1]
```diff
@@ -21,6 +21,9 @@
 @property (atomic) NSDictionary *custom;
 @property (atomic) float scoreMin;
 @property (atomic) float scoreMax;
+@property (atomic) BOOL get_all_fl;
```
Comment (jasonpeng) [inline ViSearchSDK/ViSearchAPISource/BaseSearchParams.h]: Need to follow the camel case naming convention.

File: ViSearchSDK/ViSearchAPISource/BaseSearchParams.m [hunk 2]
```diff
@@ -27,6 +30,10 @@ - (id)init
         self.custom = nil;
         self.scoreMin = 0;
         self.scoreMax = 1;
+        self.get_all_fl = false;
+        self.detection = nil;
+        
+        VALUES_DETECTION = [[NSArray alloc] initWithObjects: @"All", @"Top", @"Dress", @"Shoe", @"Bag", @"Bottom", @"Saree", nil];
```
Comment (jasonpeng) [inline ViSearchSDK/ViSearchAPISource/BaseSearchParams.m]: @yulu are we using upper case for the first letter in detection type?

Comment (yulu) [inline ViSearchSDK/ViSearchAPISource/BaseSearchParams.m]: should be lower case
@Merthur

**Comment by Merthur** [comment]
Modified codes as required @yulu @jasonpeng

**Comment by jasonpeng** [comment]
LGTM, please squash commits.
@yulu maybe you need to update sdk version.

**Comment by yulu** [comment]
approved

**Comment by yulu** [comment]
Should merge first then I pull and update the pod

**Comment by yulu** [comment]
@jasonpeng can help to merge? I do not have the permission

**Comment by jasonpeng** [comment]
Ok, I'll merge after @Merthur squashes commits (maybe into 2 commits, one for sdk change and another for example app).

**Comment by Merthur** [comment]
Hi, I have squash the commits into 2 commits @jasonpeng @yulu

**Pull request merged**